libraries/importers/component-importer/src/main/java/org/eclipse/sw360/importer/ComponentImportUtils.java

```diff
@@ -469,9 +469,21 @@ public ReleaseLinkCSVRecord apply(CSVRecord input) {
     private static Map<String, String> getVendorNameToVendorId(VendorService.Iface vendorClient) throws TException {
         Map<String, String> vendorNameToVendorId = new HashMap<>();
 
-        for (Vendor vendor : vendorClient.getAllVendors()) {
-            vendorNameToVendorId.put(vendor.getFullname(), vendor.getId());
-        }
+            for (Vendor vendor : vendorClient.getAllVendors()) {
+                if (!vendorNameToVendorId.containsKey(vendor.getShortname())) {
+                    vendorNameToVendorId.put(vendor.getShortname(), vendor.getId());
+                }
+                else {
+                    log.error("There is a clash between the shortname " + vendor.getShortname() + " and a name already present in the mapping.");
```

Comment:
Is it really an error, if you just ignore it and proceed anyway? Maybe you want to log it as a warning?

But that's more of a philosophical question than a call to action, because this module for components csv export should be removed altogether. It's a dead feature. It's slow to the point of being unusable with large amounts of data and no one uses it anyways.